### PR TITLE
chore(CI/CD): Add pr-image-builder workflow

### PR DESCRIPTION
**Pull request by davidkornel**
This PR adds a workflow that builds a Docker image based on the pull requests. It gets triggered on additional commits also. 
The image naming is `l7mp/stunner-gateway-operator:pr-<pr-number>`.

The workflow comments on the PR with the built image name to notify the user.

Currently, the run unit test job is disabled, might be useful to enable it later.

### CODE REVIEW

**Comment by davidkornel** [comment]
@rg0now As stated the current image format is `l7mp/stunner-gateway-operator:pr-<pr-number>`. I'm afraid this would result in numerous extra tags under the `l7mp/stunner-gateway-operator` image which is used for prod. Should we change the image name to `l7mp/stunner-gateway-operator-pr:<pr-number>`

**Comment by levaitamas** [comment]
Nice work! See some minor comments in the code. Two additional things: 1) IMHO, using a new registry for PR builds requires extra effort from the users, and the missing `-pr` from the name might be a source of errors. Having many tags in a single image repo is a common practice. 2) Can we add an action similar to [stunner](https://github.com/l7mp/stunner)?

Note: we should disable docker build/push actions in forks

**Comment by rg0now** [comment]
Thx, I like this. Minor wishlist item: is there any chance to garbage-collect the image at dockerhub once a PR is closed?

**Comment by davidkornel** [comment]
@levaitamas 
Alright, let's keep them under the same registry name. And we definitely should add the same action for stunner as well.

@rg0now Garbage collection is a good idea. I will add it.

File: .github/workflows/pr-image-builder.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    types: [opened, synchronize, reopened]
+
+jobs:
+  # run_unit_tests:
+  #   name: Run unit tests
+  #   uses: l7mp/stunner-gateway-operator/.github/workflows/test.yml@main
+
+  push_to_registry:
+    name: Push Docker image to DockerHub
+    runs-on: ubuntu-latest
```
[the remark this answers is not in the document]
Comment (davidkornel) [reply .github/workflows/pr-image-builder.yaml:14]: Fair enough, they won't be able to read the necessary secrets.

**Comment by davidkornel** [comment]
Testing garbage collection

**Comment by davidkornel** [comment]
Garbage collection tag results in 401. Need to figure out the issue with it and need to evaluate the response code properly.

**Comment by davidkornel** [comment]
Garbage collection is fixed
@levaitamas could you take a look?

**Review by levaitamas — APPROVED** [review]
LGTM
Thanks @davidkornel!

**Pull request merged**